File: java/ooad/TicTacToeApp/src/com/techlabs/tictactoe/Board.java
package com.techlabs.tictactoe;

import java.util.ArrayList;
import java.util.List;

public class Board {
	private ArrayList<Cell> cells=new ArrayList<Cell>();
	
	public Board(){
		initializeBoard();
	}
	
	private void initializeBoard(){
		for(int i=0;i<9;i++)
			this.cells.add(new Cell());
	}
	
	public List<Cell> getCells(){
		return this.cells;
	}
	
	public Cell getCell(int position){
		return this.cells.get(position);
	}
	
	public void displayBoard(){
		System.out.println("-----------------------------------------------\n "+this.cells.get(0)+" | "+this.cells.get(1)+" | "+this.cells.get(2)
				+"\n--- --- ---"
				+"\n "+this.cells.get(3)+" | "+this.cells.get(4)+" | "+this.cells.get(5)
				+"\n--- --- ---"
				+"\n "+this.cells.get(6)+" | "+this.cells.get(7)+" | "+this.cells.get(8));
	}
	
	public void updateBoard(MarkType mark,Integer position){
		this.cells.get(position-1).setCell(mark);
	}
	
	public boolean isBoardFilled(){
		for(Cell cell:this.cells){
			if(cell.getMark().toString().equals("E"))
				return false;
		}
		return true;
	}
	
	public Cell[][] generateBoardMatrix(){
		Cell[][] mCells=new Cell[3][3];
		
		int k=0;
		for(int i=0;i<3;i++){
			for(int j=0;j<3;j++){
				mCells[i][j]=this.cells.get(k);
				k++;
			}
		}
		return mCells;
	}
}

File: dotNet/ClassLibrary/WinForm/TicTacToeApp/TicTacToeCore/Board.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TicTacToeCore
{
    public class Board
    {
        private List<Cell> _cells = new List<Cell>();

        public Board()
        {
            InitializeBoard();
        }

        private void InitializeBoard()
        {
            for (int i = 0; i < 9; i++)
            {
                _cells.Add(new Cell());
            }
        }

        public List<Cell> Cells
        {
            get
            {
                return _cells;
            }
        }

        public Cell GetCell(int position)
        {
            return _cells.ElementAt(position);
        }

        public void DisplayBoard()
        {
            Console.WriteLine("--------------------------------------------"+
                "\n {0} | {1} | {2}"+
                "\n--- --- ---"+
                "\n {3} | {4} | {5}"+
                "\n--- --- ---"+
                "\n {6} | {7} | {8}",
                _cells.ElementAt(0), _cells.ElementAt(1), _cells.ElementAt(2),
                _cells.ElementAt(3), _cells.ElementAt(4), _cells.ElementAt(5),
                _cells.ElementAt(6), _cells.ElementAt(7), _cells.ElementAt(8));
        }

        public void UpdateBoard(MarkType mark,Int32 position)
        {
            _cells.ElementAt(position - 1).Mark = mark;
        }

        public Boolean IsBoardFilled()
        {
            foreach (Cell cell in _cells)
            {
                if (cell.Mark.Name.Equals("E"))
                {
                    return false;
                }
            }
            return true;
        }

        public Cell[,] GenerateBoardMatrix()
        {
            Cell[,] mCells = new Cell[3, 3];

            int k = 0;

            for (int i = 0; i < 3; i++)
            {
                for (int j = 0; j < 3; j++)
                {
                    mCells[i, j] = _cells.ElementAt(k);
                    k++;
                }
            }

            return mCells;
        }
    }
}
